src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,45 @@
+package duke.task;
+
+import java.time.LocalDate;
+
+import duke.Helper;
+
+/**
+ * Skeleton class for all tasks.
+ */
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Initializes an undone task with the given description.
+     * @param description description of the task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+
+    /**
+    * Marks the task as done and prints out to console that task is done.
+    */
+    public String done() {
+        this.isDone = true;
+        return Helper.formatStrings("Nice! I've marked this task as done:", this.toString());
+    }
+
+    public boolean isDescriptionContainsString(String input) {
```

Comment:
I like how you your method name accurately explains to the reader of the method's functionality. Even if it may be long, it is clear.